### PR TITLE
Comment images

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

**Comment by robbiecarlton** [comment]
Depends on https://github.com/Hylozoic/hylo-node/pull/212

**Comment by robbiecarlton** [comment]
screen shot 2017-01-30 at 3 46 56 pm

screen shot 2017-01-30 at 3 48 23 pm
screen shot 2017-01-30 at 3 49 15 pm

**Comment by robbiecarlton** [comment]
screen shot 2017-01-31 at 3 13 28 pm

File: src/components/CommentForm.js [hunk 1]
```diff
@@ -153,6 +154,7 @@ export default class CommentForm extends React.PureComponent {
             onChange={ev => this.delaySetText(ev.target.value)}
             onKeyUp={this.stopTyping}
             onKeyDown={this.handleKeyDown} />
+          <CommentImageButton postId={postId} />
```
Comment (levity) [inline src/components/CommentForm.js:157]: Let's put this behind a feature flag

**Review by levity — CHANGES_REQUESTED** [review]
Looks good to go once the feature flag is in place -- @lorenjohnson, do you want to skim over this and see if you have any opinions about style?

**Comment by lorenjohnson** [comment]
Skimmin...

File: src/components/MessageForm.js [hunk 2]
```diff
@@ -87,6 +91,7 @@ export default class MessageForm extends React.Component {
     }
 
     return <form onSubmit={this.submit} className='message-form'>
+      <CommentImageButton postId={postId} />
```
Comment (levity) [inline src/components/MessageForm.js:94]: this one too

File: src/components/Comment.js [hunk 3]
```diff
@@ -62,17 +65,29 @@ class Comment extends React.Component {
 
     if (editing) return <CommentForm commentId={comment.id} close={closeEdit} />
 
+    const { image } = comment
+
+    const encodeUrl = url => url.replace(/\//g, '%2F')
+
+    const showImage = isMobile
+      ? () => dispatch(navigate(`/image/${encodeUrl(image.url)}/${encodeUrl(location.pathname)}`))
+      : () => dispatch(showModal('image', {url: image.url}))
```
Comment (lorenjohnson) [inline src/components/Comment.js:74]: I'd personally prefer:
```javascript
showImage = () => {
  if (isMobile) {
    dispatch(navigate(`/image/${encodeUrl(image.url)}/${encodeUrl(location.pathname)}`))
  } else {
    dispatch(showModal('image', {url: image.url}))
  }
}
```
... or similar. I'm just not a fan of the ternary operator for multi-line anythings.

File: src/components/CommentImageButton.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    dispatch(uploadImage(imageUploadSettings(currentUser.id, postId)))
+    .then(({ payload, error }) => {
+      if (error) return dispatch(notify('There was a problem sending your image. Please try again in a moment', {type: 'error'}))
+      return dispatch(createComment({postId, imageUrl: payload}))
+    })
+
+  return <a className='send-image' onClick={sendImage}><Icon name='Camera' /></a>
+}
+CommentImageButton.contextTypes = {
+  dispatch: React.PropTypes.func, currentUser: React.PropTypes.object
```
Comment (lorenjohnson) [inline src/components/CommentImageButton.js:23]: Do this above:
```javascript
import React, { PropTypes } from 'react'
```
and then here: 
```javascript
 dispatch: PropTypes.func, ...
```
I prefer that both to what you've done here and to fully resolving down to just "func".  Again, just my own opinions.

File: src/components/CommentImageButton.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+export default CommentImageButton
```
Comment (lorenjohnson) [inline src/components/CommentImageButton.js]: You could also just do a:
```javascript
export default const CommentImageButton
```
above...

**Review by lorenjohnson — APPROVED** [review]
LGTM

**Pull request merged**